packages/central-server/src/apiV2/export/exportOptionSet/exportOptionSet.js

```diff
@@ -0,0 +1,46 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
+ */
+
+import xlsx from 'xlsx';
+import { respondWithDownload, toFilename } from '@tupaia/utils';
+import { getExportPathForUser } from '../getExportPathForUser';
+
+const objectToHumanReadableKeyValuePairs = obj => {
+  const entries = [];
+  for (const [key, value] of Object.entries(obj)) {
+    entries.push(`${key}: ${value}`);
+  }
+  return entries.join('\n');
+};
+
+export async function exportOptionSet(req, res) {
+  const { models, userId } = req;
+  const { optionSetId } = req.params;
+
+  const optionSetRecord = await models.optionSet.findOne({ id: optionSetId });
+  const optionRecords = await optionSetRecord.options();
+  const options = optionRecords.map(option => {
+    return {
+      value: option.value,
+      label: option.label,
+      sort_order: option.sort_order,
+      attributes: objectToHumanReadableKeyValuePairs(option.attributes),
+    };
+  });
+
+  const worksheet = xlsx.utils.json_to_sheet(options);
+  const sheetName = optionSetRecord.name.substring(0, 31); // Sheet name max. length is hard-coded in Excel
+
+  // Upgrade xlsx to v0.20 or newer to do next two lines in one go
```

Comment:
There are a couple of [open Dependabot alerts](https://github.com/beyondessential/tupaia/security/dependabot?q=package%3Axlsx+manifest%3Ayarn.lock+has%3Apatch) to do with the version of SheetJS we’re using; but when I tried updating it as part of RN-1112 it did break a few things

At the time I left it since there were only moderate alerts, but more recently a high severity one’s come up—not sure if it’s worth revisiting now

Reply:
Probably worth making a separate ticket for that 🙏

Reply:
Putting ticket here for reference: RN-1310